File: src/main/java/me/Geek/Libs/DataBase/sqlite.java
package me.Geek.Libs.DataBase;

import com.zaxxer.hikari.HikariDataSource;
import me.Geek.Configuration.ConfigManager;

import java.io.File;
import java.sql.Connection;
import java.sql.SQLException;
import java.sql.Statement;

/**
 * 作者: 老廖
 * 时间: 2022/7/23
 **/
public class sqlite extends DataSub{

    private HikariDataSource SQLITE;
    public sqlite() {
        super();
    }

    @Override
    public void onLoad() {
        final String SqliteUrl = "jdbc:sqlite:" + ConfigManager.plugin.getDataFolder() + File.separator +"GeekData.db";
        SQLITE = new HikariDataSource();
        SQLITE.setDataSourceClassName("org.sqlite.SQLiteDataSource");
        SQLITE.addDataSourceProperty("url", SqliteUrl);
        //附件参数
        SQLITE.setMaximumPoolSize(ConfigManager.MAXIMUM_POOL_SIZE);
        SQLITE.setMinimumIdle(ConfigManager.MINIMUM_IDLE);
        SQLITE.setMaxLifetime(ConfigManager.MAXIMUM_LIFETIME);
        SQLITE.setKeepaliveTime(ConfigManager.KEEPALIVE_TIME);
        SQLITE.setConnectionTimeout(ConfigManager.CONNECTION_TIMEOUT);
        SQLITE.setPoolName("GeekMail-SQLITE");
        createSqliteTables();
    }

    @Override
    public Connection getConnection() throws SQLException {
        return SQLITE.getConnection();
    }

    @Override
    public void onStop() {
        if (SQLITE != null) SQLITE.close();
    }

    private void createSqliteTables() {
        try (Connection connection = getConnection()) {
            try (Statement statement = connection.createStatement()){
                statement.execute("PRAGMA foreign_keys = ON;");
                statement.execute("PRAGMA encoding = 'UTF-8';");
                statement.execute("CREATE TABLE IF NOT EXISTS `mail_data` (" +
                        " `id` integer PRIMARY KEY, " +
                        " `mail_id` CHAR(36) NOT NULL, " +
                        " `state` VARCHAR(256) NOT NULL, " +
                        " `type` text NOT NULL, " +
                        " `sender` CHAR(36) NOT NULL, " +
                        " `target` CHAR(36) NOT NULL, " +
                        " `title` text NOT NULL, " +
                        " `text` longtext NOT NULL, " +
                        " `money` double NOT NULL DEFAULT '0', " +
                        " `points` integer NOT NULL DEFAULT '0', " +
                        " `exp` integer NOT NULL DEFAULT '0', " +
                        " `item` longtext NOT NULL, " +
                        " `commands` longtext NOT NULL);");
            }
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}

File: Scheduler/src/main/kotlin/me/geek/mail/scheduler/sql/Sqlite.kt
package me.geek.mail.scheduler.sql

import com.zaxxer.hikari.HikariDataSource
import java.io.File
import java.sql.Connection

/**
 * 作者: 老廖
 * 时间: 2022/10/16
 *
 **/
class Sqlite(
    private val config: SqlConfig
): SqlService() {
    override var isActive: Boolean = false
    private val dataSource by lazy {
        HikariDataSource().apply {
            maximumPoolSize = config.hikari_settings.maximum_pool_size
            minimumIdle = config.hikari_settings.minimum_idle
            maxLifetime = config.hikari_settings.maximum_lifetime.toLong()
            keepaliveTime = config.hikari_settings.keepalive_time.toLong()
            connectionTimeout = config.hikari_settings.connection_timeout.toLong()
            poolName = "GeekMail-SQLITE"
        }
    }

    override fun getConnection(): Connection {
        return this.dataSource.connection
    }

    override fun onStart() {
        val url = "jdbc:sqlite:" + config.sqlite + File.separator + "data.db"
        dataSource.dataSourceClassName = "org.sqlite.SQLiteDataSource"
        dataSource.addDataSourceProperty("url", url)
        isActive = testSql()
    }

    override fun onClose() {
        this.dataSource.close()
    }
}